openwisp_radius/base/models.py

```diff
@@ -899,7 +899,7 @@ def prefix_add(self, prefix, n, password_length=BATCH_DEFAULT_PASSWORD_LENGTH):
             user.full_clean()
             self.save_user(user)
         pdf_file = generate_pdf(prefix, {'users': user_password})
-        pdf_file.name = pdf_file.name.split('/')[-1]
+        pdf_file.name = f'{settings.MEDIA_ROOT}/{prefix}.pdf'
```

Comment:
I have tested this now, I get a 404 if I try to open the link to the file in the admin, so there's more work to do here.
I would go back to the idea of deleting the file if validation goes wrong.

Another issue that I found is that creating the file in `/tmp` may not spot if there's already a file with the same name.

Reply:
> I have tested this now, I get a 404 if I try to open the link to the file in the admin, so there's more work to do here.

Fixed it.

> Another issue that I found is that creating the file in /tmp may not spot if there's already a file with the same name.

It doesn't matter if the old file is overridden does it?
In anycase, I am already working on #102 so I suppose it shouldn't matter soon! :smile:

Reply:
Ok let's move on, #102 is the best solution